guidelines/CPP_STYLE_GUIDE.md

```diff
@@ -0,0 +1,221 @@
+# AICA C++ Style Guide and Recommendations
+
+* [AICA Style Summary](#AICA-style-summary)
+* * [General](#general)
+* * [File name and structure](#file-name-and-structure)
+* * [General naming](#general-naming)
+* * [Format and whitespace](#format-and-whitespace)
+* [Automatic formatting](#automatic-formatting)
+* * [clang-format](#clang-format-installation-and-usage)
+* * [CLion](#clion)
+* * [Visual Studio Code](#visual-studio-code)
+* * [Sublime Text](#sublime-text)
+
+
+## Foreword 
+C++ is a beast of a language - it is traditional, and it is modern. It is so close, and yet so far, from base C. It lets you get incredibly close to the hardware and manage every bit of memory yourself, and at the same time it provides incredible abstraction patterns.
+
+Within AICA, where people's experience and ambitions with coding projects are diverse, the aim of this page is not to militantly enforce modern, gold-standard programming. Rather, we just want code to be consistent and readable so we can focus on actual research problems.
+
+The [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html) has been widely adopted as good and safe practice. [It is not without its flaws](https://eyakubovich.github.io/2018-11-27-google-cpp-style-guide-is-no-good/), and can be quite restrictive particularly towards modern use of the language. Still, its rules are clear and provide a solid foundation for shared code projects.
+
+The [ANYbotics ANYmal Research C++ Style Guide](https://anybotics.github.io/styleguide/cppguide.html) makes a few minor adaptations to the base Google style in the context of robotics research, and is the recommended reference for AICA.
+
+To further promote readability and usability within AICA, some additional minor changes have been suggested below. **Read and adopt the summary below first, then use the ANYbotics guide for everything else.**
+
+Use this guide in conjunction with git best practices and pull request guidelines and aim for incremental improvement to your code quality and usability with each new commit.
```

Comment:
nitpick: There were probably some links here previously, do you  want to keep that part?

Reply:
I removed them intentionally since we don't have them hosted anywhere (yet). I would suggest eventually adding such guides to the same folder in this repository. That way it's centralized and public (which helps with regards to the open source parts of the stack)